services/multiinstance/staticdelay/shouldpropose.go

```diff
@@ -0,0 +1,60 @@
+// Copyright © 2024 Attestant Limited.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package staticdelay
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/attestantio/go-eth2-client/api"
+	"github.com/attestantio/vouch/services/beaconblockproposer"
+)
+
+// ShouldPropose returns true if this Vouch instance should propose.
+func (s *Service) ShouldPropose(ctx context.Context, duty *beaconblockproposer.Duty) bool {
+	if duty == nil {
+		s.log.Debug().Msg("No duty supplied")
+
+		return false
+	}
+
+	log := s.log.With().Uint64("slot", uint64(duty.Slot())).Logger()
+	log.Trace().Msg("Checking if we should propose")
+
+	if s.proposerActive.Load() {
+		log.Trace().Msg("We are the active proposer")
+		s.enableProposer(ctx)
```

Comment:
Do we not just need to call `monitorActive` here?

Reply:
As per above.